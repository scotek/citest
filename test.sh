#!/bin/bash
# Simple (and horrible) shell script to test some input and output
# Use a proper test framework if you are doing this for real.

# Run the program with the dummy input and save the output
./prog <test.input >test.real

# Compare the actual output with the expected output with the "diff" command
df=`diff test.real test.expected`

# If they are different, show failed test info and exit with an error code
if [ $? != 0 ]
then
    echo "FAIL!"
    echo "$df"
    exit 1
fi
