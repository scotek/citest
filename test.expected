Paul's great reversing program!
Please enter up to 99 characters: Reversed is: gfedcba
